verification/compare.py

```diff
@@ -296,11 +301,35 @@ def main(file1, file2, allowed_mismatch_columns):
             for sample in issue["samples"]:
                 msg += f"\n    - Index {sample['index']}: Left = {sample['left']}, Right = {sample['right']}"
         print(msg)
-    issues_leading_to_failure = [
-        issue
-        for issue in issues
-        if issue["column"] not in allowed_mismatch_columns.split(",")
-    ]
+    # Parse option lists
+    allowed_mismatch_set = set(c.strip() for c in allowed_mismatch_columns.split(",") if c.strip())
+    ignore_missing_set = set(c.strip() for c in ignore_missing_columns.split(",") if c.strip())
+    forbidden_set = set(c.strip() for c in forbidden_columns.split(",") if c.strip())
+
+    # Check for forbidden columns
+    if forbidden_set:
+        fields1 = get_all_field_names(table1.schema)
+        fields2 = get_all_field_names(table2.schema)
+        found_forbidden = (fields1 | fields2) & forbidden_set
+        if found_forbidden:
+            print(f"\n{'=' * 70}")
+            print(f"FORBIDDEN COLUMNS CHECK FAILED")
+            print(f"{'=' * 70}")
+            print(f"Found forbidden columns: {sorted(found_forbidden)}")
+            exit(1)
+
+    # Filter issues
+    issues_leading_to_failure = []
```

Comment:
I think moving this up before the comparison is printed would be good. So that we either don't print explicitly allowed mismatches at all, or just print them with an extra info that these are checked and therefore allowed. I leave the concrete implementation here up to you.